### PR TITLE
Run dynamic drive discovery service as separate container.

### PR DESCRIPTION
**Pull request by balamurugana**
Signed-off-by: Bala.FA <bala@minio.io>

### CODE REVIEW

File: pkg/node/uevent.go [hunk 1]
```diff
@@ -418,3 +418,28 @@ func (handler *ueventHandler) processLoop(ctx context.Context) {
 		handler.processEvent(ctx, device, event["ACTION"])
 	}
 }
+
+func StartDynamicDriveDiscovery(ctx context.Context, identity, nodeID, rack, zone, region string, loopbackOnly bool) error {
+	handler := &ueventHandler{
+		nodeID: nodeID,
+		topology: map[string]string{
```
Comment (wlan0) [inline pkg/node/uevent.go:425]: do we need topology?

File: pkg/installer/daemonset.go [hunk 2]
```diff
@@ -146,7 +146,7 @@ func createDaemonSet(ctx context.Context, c *Config) error {
 						args = append(args, "--loopback-only")
 					}
 					if c.DynamicDriveDiscovery {
-						args = append(args, "--dynamic-drive-discovery")
+						args = append(args, "--enable-dynamic-discovery")
```
Comment (Praveenrajmani) [inline pkg/installer/daemonset.go:149]: should we support backward compatility here? renaming this flag will break backward compatibility. 

If v2.0.1 client is used to install, `--dynamic-drive-discovery` will be set on the binary.

File: pkg/node/node.go [hunk 3]
```diff
@@ -93,32 +93,6 @@ func NewNodeServer(ctx context.Context, identity, nodeID, rack, zone, region str
...
-				return nil, err
-			}
-		}
-
-		klog.V(3).Info("Doing initial drive sync up")
-		handler.syncDrives(ctx)
-
-		// Start background tasks
-		go handler.processLoop(ctx)
-	}
```
Comment (Praveenrajmani) [inline pkg/node/node.go:120]: if we decide to support backward compatibility, we need to keep this code. we need to make this code segment run, if `--dynamic-drive-discovery` flag is set.

File: pkg/node/uevent.go (hunk 1 above)
Comment (Praveenrajmani) [reply pkg/node/uevent.go:425]: yes, we need topology as uevent handler creates new drive objects if any drives are added

File: pkg/installer/daemonset.go (hunk 2 above)
Comment (wlan0) [reply pkg/installer/daemonset.go:151]: yeah, lets keep the existing flag

File: pkg/node/node.go (hunk 3 above)
Comment (wlan0) [reply pkg/node/node.go:120]: It is going to run as a separate container. This is not needed.

**Comment by wlan0** [comment]
LGTM, only one minor comment about flag. please fix, and it can be merged

**Review by wlan0 — APPROVED** [review]

**Pull request merged**